src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public readonly a: number,
+    public readonly b: number,
+    public readonly c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('error length');
```

Comment:
It is better to describe errors more informatively.
```suggestion
      throw new Error('All sides of the triangle must be greater than zero'');
```